test/Microsoft.ML.Functional.Tests/Explainability.cs

```diff
@@ -0,0 +1,284 @@
+    /// Test explainability features.
+    /// </summary>
+    public class Explainability : BaseTestClass
+    {
+        public Explainability(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        /// <summary>
+        /// GlobalFeatureImportance: PFI can be used to compute global feature importance.
+        /// </summary>
+        [Fact]
+        public void GlobalFeatureImportanceWithPermutationFeatureImportance()
+        {
+            var mlContext = new MLContext(seed: 1, conc: 1);
+
+            // Get the dataset
+            var data = mlContext.Data.ReadFromTextFile<HousingRegression>(GetDataPath(TestDatasets.housing.trainFilename), hasHeader: true);
+
+            // Create a pipeline to train on the housing data.
+            var pipeline = mlContext.Transforms.Concatenate("Features", HousingRegression.Features)
+                .Append(mlContext.Regression.Trainers.StochasticDualCoordinateAscent());
+
+            // Fit the pipeline and transform the data.
+            var model = pipeline.Fit(data);
+            var transformedData = model.Transform(data);
+
+            // Compute the permutation feature importance to look at global feature importance.
+            var permutationMetrics = mlContext.Regression.PermutationFeatureImportance(model.LastTransformer, transformedData);
+
+            // Make sure the correct number of features came back.
+            Assert.Equal(HousingRegression.Features.Length, permutationMetrics.Length);
+            foreach (var metricsStatistics in permutationMetrics)
+                Common.AssertMetricsStatistics(metricsStatistics);
+        }
+
+        /// <summary>
+        /// GlobalFeatureImportance: A linear model's feature importance can be viewed through its weight coefficients.
+        /// </summary>
+        /// <remarks>
+        /// Note that this isn't recommended, as there are quite a few statistical issues with interpreting coefficients
+        /// as weights, but it's common practice, so it's a supported scenario.
+        /// </remarks>
+        [Fact]
+        public void GlobalFeatureImportanceForLinearModelThroughWeights()
+        {
+            var mlContext = new MLContext(seed: 1, conc: 1);
+
+            // Get the dataset
+            var data = mlContext.Data.ReadFromTextFile<HousingRegression>(GetDataPath(TestDatasets.housing.trainFilename), hasHeader: true);
+
+            // Create a pipeline to train on the housing data.
+            var pipeline = mlContext.Transforms.Concatenate("Features", HousingRegression.Features)
+                .Append(mlContext.Regression.Trainers.StochasticDualCoordinateAscent());
+
+            // Fit the pipeline and transform the data.
+            var model = pipeline.Fit(data);
+            var linearModel = model.LastTransformer.Model;
+
+            // Make sure the number of model weights returned matches the length of the input feature vector.
+            var weights = linearModel.Weights;
```

Comment:
>Weights [](start = 38, length = 7)

validate that weights are reasonable with baseline or some other heuristic, not that they are just nonnegative

Reply:
See above comment about FunctionalTests vs. Correctness tests and Baseline tests.

---
In reply to: [258590457](https://github.com/dotnet/machinelearning/pull/2584#discussion_r258590457) [](ancestors = 258590457)